src/plugins/blackbox/public/js/blackbox.js

```diff
@@ -616,10 +616,55 @@
       
     }.bind(null, options.collection));
   };
+
+  Blackbox.prototype._exportVideoToNativeFileSystem = function _exportVideoToNativeFileSystem(options,callback){
```

Comment:
This function is much more efficient that the web only export as it processes a single frame at a time and streams it to the filessystem.